frontend/src/components/pages/shadowlinks/details/shadowlink-details-page.tsx

```diff
@@ -0,0 +1,246 @@
+ */
+
+'use client';
+
+import { Button } from 'components/redpanda-ui/components/button';
+import { Text } from 'components/redpanda-ui/components/typography';
+import { AlertCircle, Loader2, Trash2 } from 'lucide-react';
+import { runInAction } from 'mobx';
+import { useEffect, useMemo, useState } from 'react';
+import {
+  useDeleteShadowLinkMutation,
+  useFailoverShadowLinkMutation,
+  useGetShadowLinkQuery,
+  useListShadowTopicInfiniteQuery,
+} from 'react-query/api/shadowlink';
+import { useNavigate, useParams } from 'react-router-dom';
+import { toast } from 'sonner';
+import { uiState } from 'state/ui-state';
+
+import { DeleteShadowLinkDialog } from './delete-shadowlink-dialog';
+import { FailoverDialog } from './failover-dialog';
+import { ShadowLinkDiagram } from './shadow-link-diagram';
+import { ShadowLinkMetrics } from './shadow-link-metrics';
+import { ShadowTopicsTable } from './shadow-topics-table';
+
+// Update page title using uiState pattern
+export const updatePageTitle = (shadowLinkName: string) => {
+  runInAction(() => {
+    uiState.pageTitle = shadowLinkName;
+    uiState.pageBreadcrumbs = [
+      { title: 'Shadow Links', linkTo: '/shadowlinks' },
+      { title: shadowLinkName, linkTo: `/shadowlinks/${shadowLinkName}` },
+    ];
+  });
+};
+
+export const ShadowLinkDetailsPage = () => {
+  const { name } = useParams<{ name: string }>();
+  const navigate = useNavigate();
+  const [showDeleteDialog, setShowDeleteDialog] = useState(false);
+  const [showFailoverDialog, setShowFailoverDialog] = useState(false);
+  const [failoverTopicName, setFailoverTopicName] = useState<string>('');
+  const [topicNameFilter, setTopicNameFilter] = useState('');
+
+  // Update page title
+  useEffect(() => {
+    if (name) {
+      updatePageTitle(name);
+    }
+  }, [name]);
+
+  // Fetch shadow links data
+  const { data: shadowLinksData, isLoading, error: errorGetShadowLink, refetch } = useGetShadowLinkQuery({ name });
+  const {
+    data: shadowTopicsData,
+    refetch: refetchTopics,
+    fetchNextPage,
+    hasNextPage,
+    isFetchingNextPage,
+    isFetching: isFetchingTopics,
+  } = useListShadowTopicInfiniteQuery(
+    {
+      shadowLinkName: name,
+      filter: topicNameFilter
+        ? {
+            topicNameContains: topicNameFilter,
+          }
+        : undefined,
+      pageSize: 100,
+    },
+    { refetchInterval: 5000 }
+  );
+
+  // Find the specific shadow link by name
+  const shadowLink = shadowLinksData?.shadowLink;
+
+  // Flatten all pages of topics from infinite query
+  const topics = useMemo(() => shadowTopicsData?.pages?.flatMap((page) => page.shadowTopics) ?? [], [shadowTopicsData]);
+
+  const { mutate: deleteShadowLink, isPending: isDeleting } = useDeleteShadowLinkMutation({
+    onSuccess: () => {
+      toast.success(`Shadowlink ${name} deleted`);
+      navigate('/shadowlinks');
+    },
+    onError: (error) => {
+      toast.error(`Failed to delete shadowlink: ${error.message}`);
```

Comment:
Let's use the `formatToastErrorMessageGRPC` utility function for all connect errors